### PR TITLE
readme: be more specific, explain what one gets

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

**Comment by ThomasWaldmann** [comment]
@sunknudsen ^^^

**Comment by sunknudsen** [comment]
@ThomasWaldmann Thanks for the PR. `borgbackup-llfuse` [depends](https://github.com/borgbackup/homebrew-tap/blob/master/Formula/borgbackup-llfuse.rb#L17) on [FUSE for macOS](https://osxfuse.github.io/) therefore when using this formula vs [borgbackup](https://formulae.brew.sh/formula/borgbackup), one has to install the dependency using `brew install --cask osxfuse` first.

This dependency was previously avoided given `brew install borgbackup` was packaged as a bottle automatically by Homebrew (vs `borgbackup-llfuse` which installs from source).

Would you like me to edit your PR?

**Comment by ThomasWaldmann** [comment]
Sure, edit it! :-)

**Comment by sunknudsen** [comment]
@ThomasWaldmann Edit done... before getting this project out to the world, perhaps we should consider renaming the formula to `borgbackup-fuse`?

**Comment by sunknudsen** [comment]
More changes upstream... `depends_on :osxfuse` will be deprecated shortly (see https://github.com/Homebrew/brew/issues/9401#issuecomment-738007106).

**Comment by sunknudsen** [comment]
@m3nu 👆

**Comment by m3nu** [comment]
Depending on Casks seems to be a mess either way. I'm looking at some alternatives.

**Comment by sunknudsen** [comment]
I agree... depending on casks is essentially not supported anymore by Homebrew. That being said, installing `osxfuse` using `brew install --cask osxfuse` before running `brew install borgbackup-llfuse` (which I believe should be renamed to `borgbackup-fuse`) works fine.

**Comment by ThomasWaldmann** [comment]
> renaming the formula to |borgbackup-fuse|?

Not sure.

In borg 1.2 you need either llfuse or pyfuse3.

If both shall be offered, keeping -llfuse in the name would be good.

Alternatively, only offer 1 way and decide it for the user (then it 
could be -fuse).

**Comment by m3nu** [comment]
Since the formulas conflict anyways, why not just call it the same? Docs only say:

> If your formulae have the same name as Homebrew/homebrew-core formulae they cannot be installed side-by-side.[1](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap#naming-your-formulae-to-avoid-clashes)

**Comment by sunknudsen** [comment]
I don't dislike the fact open source purists (I don't mean that in a negative way) may choose to install `borgbackup` over `borgbackup-fuse` as the latter depends on proprietary code. My gut feeling is using a different name might help avoid confusion.

**Comment by sunknudsen** [comment]
@ThomasWaldmann

> If both shall be offered, keeping -llfuse in the name would be good.

I agree and that is why I used the `llfuse` suffix in the first place. That being said, given `llfuse` is deprecated, my gut feeling is we might want to switch to `pyfuse3` when possible (see https://github.com/libfuse/pyfuse3/issues/32).

**Comment by ThomasWaldmann** [comment]
we'll start trying pyfuse3 after 1.2 is out. if it is available good enough, works good enough and everybody is happy, we can drop llfuse.

File: README.md [hunk 1]
```diff
@@ -1,13 +1,20 @@
 # BorgBackup [Homebrew](https://brew.sh/) tap
 
-## How do I install formulas?
+## Why install [Borg](https://www.borgbackup.org/) using this tap vs `brew install borgbackup`?
+
+The `borgbackup-llfuse` formula maintained in this tap depends on [osxfuse](https://formulae.brew.sh/cask/osxfuse) and [llfuse](https://github.com/python-llfuse/python-llfuse/) which are required to mount repositories or archives using `borg mount`.
+
+These dependencies have been [removed](https://github.com/Homebrew/homebrew-core/commit/8c2f17e3b653347ada86d353243e2d6b6cb10fda#diff-4a25217474a5eb61d0776ab4cabc43b42689bc7b3efaaed400f799631dcec71f) from Homebrew’s [borgbackup](https://formulae.brew.sh/formula/borgbackup) formula because [FUSE for macOS](https://osxfuse.github.io/) is no longer open source.
```
Comment (ThomasWaldmann) [inline README.md:7]: here it talks about fuse for macos, above it is named osxfuse - migh be confusing for people who do not know it is referring to the same project.

**Comment by sunknudsen** [comment]
> we'll start trying pyfuse3 after 1.2 is out. if it is available good enough, works good enough and everybody is happy, we can drop llfuse.

Got it... I recommend renaming the formula to `borgbackup-fuse` before the formula is made public... I am planning on publishing an episode tomorrow so ideally we would be set on the name today. @ThomasWaldmann @m3nu Thoughts? If we proceed with the name change, I would simply rename the formula... aside from us, I don't think others will have had a chance to switch from `borgbackup`.

File: README.md (hunk 1 above)
Comment (sunknudsen) [reply README.md:7]: @ThomasWaldmann I agree the naming convention of Homebrew vs the repo vs the website is confusing... to make things worse, FUSE for macOS has been renamed to macFUSE starting at version 4. What do you suggest? I guess we could use `osxfuse` as this is the name of the formula?

**Comment by ThomasWaldmann** [comment]
i put it yesterday on twitter/mastodon...

**Comment by ThomasWaldmann** [comment]
i'm fine with both pkg names, choose what you like. :-)

**Comment by sunknudsen** [comment]
> i put it yesterday on twitter/mastodon...

@ThomasWaldmann I guess we should deprecate `borgbackup-llfuse` and keep both for a few months then? Thanks for your guidance btw... This is my second collab on an open source project with considerable reach... I want to make sure I don't make naive mistakes.

**Comment by ThomasWaldmann** [comment]
i just pointed users to the ticket (which points to the repo here).

so guess it's better to just change it, should not affect many (and even if, it doesn't matter if it breaks now or in a few months).

**Comment by sunknudsen** [comment]
@m3nu We appear to have consensus to rename the package to `borgbackup-fuse`. Cool?

**Comment by sunknudsen** [comment]
Standby for @m3nu's feedback. PR is ready for final review... I also fixed the `depends_on :osxfuse` deprecation issue.

**Comment by m3nu** [comment]
Either name is fine, as long as the product works.

I'll try the update today. We may also want to run `brew test` via Github actions.

**Comment by sunknudsen** [comment]
@ThomasWaldmann Ready to merge? @m3nu will likely submit another PR with tests and other optimizations but reports the formula also worked at his end (see https://github.com/borgbackup/borg/issues/5522#issuecomment-738508169).

**Comment by m3nu** [comment]
We can do it like that. I based everything off this branch. So can merge my commits on top of it. It [adds](https://github.com/m3nu/homebrew-tap):

- Lint formula to pass `brew audit`
- Add back tests to formula
- Run install and tests in GH actions.
- Add some extra tests for `borg mount`

**Comment by sunknudsen** [comment]
Nice additions @m3nu. Never used GitHub actions. Looks promising!

**Comment by sunknudsen** [comment]
@ThomasWaldmann What governance would you like us to follow in the context of merging PRs? This PR is ready to go and I would like to start recording an episode so ideally we would merge it shortly.

**Comment by ThomasWaldmann** [comment]
4768151e4ee4515f232a76dfe89488b0672d840b seems to rename a file **and** do significant changes to the contents of that file (the ruby code of the formula).

That is almost never a good idea because it is hard to see. If something needs to be renamed, only do the rename and commit.

Besides that, it is unrelated to the scope of this PR, which was to improve the readme.

File: README.md (hunk 1 above)
Comment (ThomasWaldmann) [reply README.md]: it could at both places call it osxfuse.

at first place it could also say `(aka FUSE for macOS)`.

**Comment by sunknudsen** [comment]
@ThomasWaldmann I have to admit this initial PR got messy. Sorry about that. What is the best way to split it into multiple PRs?

**Comment by ThomasWaldmann** [comment]
Maybe like this:

- keep a copy of the modified formula at some other place.
- copy the unmodified formula over the renamed one and then git add / git commit --amend / git push -f or so.
- finish this PR and modify the formula in another PR as needed.

**Comment by sunknudsen** [comment]
@ThomasWaldmann OK, I will try that... on it!

**Comment by sunknudsen** [comment]
@ThomasWaldmann I believe I cleaned up the PR mess... I feel like such a newbie... This was my first time creating conflicting pull requests all at the same time. Looking forward to your feedback.

**Comment by ThomasWaldmann** [comment]
looks good, merged!

**Pull request merged**